components/sup/src/manager/service_updater.rs

```diff
@@ -384,67 +445,100 @@ impl Worker {
     /// Polls until a newer version of the specified package is
     /// available. When such a package is found, it is installed, and
     /// the function exits.
-    fn run_once(&mut self, sender: SyncSender<PackageInstall>, ident: PackageIdent) {
+    fn run_once(
+        &mut self,
+        sender: Sender<PackageInstall>,
+        ident: PackageIdent,
+        kill_rx: Receiver<bool>,
+    ) {
         // Fairly certain that this only gets called in a rolling update
         // scenario, where `ident` is always a fully-qualified identifier
         outputln!("Updating from {} to {}", self.current, ident);
         let install_source = (ident, *PackageTarget::active_target()).into();
-        loop {
-            let next_time = self.next_period_start();
+        let mut next_time = None::<SteadyTime>;
```

Comment:
I think if you initially set `next_time` to "now", you'll get the same ultimate effect, without the verbosity of using an `Option` here.

```suggestion
        let mut next_time = SteadyTime::now();
```

Reply:
Ah, good call.